### ISSUE TITLE
Model Training is Behind in Performance

### ISSUE THREAD

**Issue by stanleyshly** [facebookresearch/3detr#16]
When training with one batch, and scaling all training rate parameters(except the weight decay) by the sqrt(1/8), I notice my model's training performance is way behind the checkpoints(90 epoch, 180 epochs, 360 epochs, 720 epochs, etc), but a significant amounts by having the results of mAP that is often 40% to 60% of the mAP of the checkpoints.
I also tried gradient accumulation up to a batch size of 8, and used the included training parameters, but my model performance, even when training with the scannet_quick script, I still only have a mAP of 30-35. 
Any tips for getting full performance with a batch size of 1, for setups with limited memory?

**Comment by pummi823** [comment]
I train the sunrgbd dataset and meet the same question as yours, the mAPs are very low which around 30% when the epochs are 100+

**Comment by stanleyshly** [comment]
> I train the sunrgbd dataset and meet the same question as yours, the mAPs are very low which around 30% when the epochs are 100+

I see, what hardware are you using for training? I have a tad of a limited setup, with a 1060 3gb, meaning I can only do one batch at a time. What is your batch size when training?

**Comment by pummi823** [comment]
MY: --dataset_num_workers 4, --batchsize_per_gpu 4,

**Comment by pummi823** [comment]
But, I tested the PTH file provided by the author, and the results were much higher than those published by him!!

**Comment by stanleyshly** [comment]
> But, I tested the PTH file provided by the author, and the results were much higher than those published by him!!

Interesting, my results with the test script on the scannet dataset are around similar to the claimed results. Okay, so batchsize probably isn't the issue, since your batch size of 4 is a lot closer to the trained 8, yet your results are approximately in the same ball park as mine.

**Comment by pummi823** [comment]
I have saw the other question in this project, and the author said that the learning rate is important, and that questioner did get much better results when he used the lower lr, maybe we can try

**Comment by stanleyshly** [comment]
That is true, I am scaling my lr by sqrt{batch_size_now/8}, maybe we could try a more linear scaling?

**Comment by pummi823** [comment]
Yes, let us try, and we can keep communicating if anyone finds an effective way

**Comment by stanleyshly** [comment]
Alright, will do. My training is really slow though, 3-4 days for 1080 epochs, hopefully the authors will step in and point out what we are doing wrong.

**Comment by imisra** [comment]
Hi @stanleyshly, @pummi823 

I haven't tried training at the batchsizes you report. I will do that and get back to you. I suspect it is a matter of finding the right hyperparameters for this setup.
